### PR TITLE
Clean up ECCO functionality

### LINKED ISSUE #118: `ecco` or `ECCO`?
Sometimes we call it `ECCO` and sometimes we call it `ecco`:

https://buildkite.com/clima/climaocean-ci/builds/1066#01912ed9-44af-4487-af5b-15a09d3ed293/202-820

Like, the module `ECCO` but then `ecco_mask` or `empty_ecco_field`:

https://github.com/CliMA/ClimaOcean.jl/blob/50661675ccf86e143ef33b53b5f07c170a8fb1f5/src/DataWrangling/ECCO.jl#L1

https://github.com/CliMA/ClimaOcean.jl/blob/50661675ccf86e143ef33b53b5f07c170a8fb1f5/src/DataWrangling/ECCO.jl#L81

We should try to decide how this word is supposed to be capitalized and stick with that spelling. It's not like a typical word that can be capitalized two ways.

It seems like it should always be `ECCO` to me since it's an acronym (Estimating the Climate and Circulation of the Ocean).

### LINKED ISSUE #165: Why is `ecco_restoring.jl` included at the end of the `ECCO.jl` file?
It's not the normal pattern we like to use

https://github.com/CliMA/ClimaOcean.jl/blob/dd9148a8f702699ddf2947d3a1a094adddbd9bb1/src/DataWrangling/ECCO.jl#L312

### PR DESCRIPTION
**Pull request by glwagner**
This cleans up how the ECCO stuff is implemented. 

Closes #118 
Closes #165 

It would be nice to also address #163.

### CODE REVIEW

File: src/ClimaOcean.jl [hunk 1]
```diff
@@ -8,15 +8,15 @@ export
     JRA55_prescribed_atmosphere,
     JRA55NetCDFBackend,
     ECCOMetadata,
-    ecco2_field,
+    ECCO2_field,
```
Comment (simone-silvestri) [inline src/ClimaOcean.jl:11]:
```suggestion
```

File: src/DataWrangling/ECCO/ECCO_metadata.jl [hunk 2]
```diff
@@ -122,9 +131,9 @@ ecco_location = Dict(
 )
 
 # URLs for the ECCO datasets specific to each version
-urls(::ECCOMetadata{<:Any, <:ECCO2Monthly}) = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/monthly/"
-urls(::ECCOMetadata{<:Any, <:ECCO2Daily})   = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/daily/"
-urls(::ECCOMetadata{<:Any, <:ECCO4Monthly}) = "https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/interp_monthly/"
+urls(::ECCOMetadata{<:Any, <:ECCO2Monthly}) = "https://ECCO.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/monthly/"
+urls(::ECCOMetadata{<:Any, <:ECCO2Daily})   = "https://ECCO.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/daily/"
+urls(::ECCOMetadata{<:Any, <:ECCO4Monthly}) = "https://ECCO.jpl.nasa.gov/drive/files/Version4/Release4/interp_monthly/"
```
Comment (simone-silvestri) [inline src/DataWrangling/ECCO/ECCO_metadata.jl:136]: I am not sure websites are case sensitive or not
```suggestion
urls(::ECCOMetadata{<:Any, <:ECCO2Monthly}) = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/monthly/"
urls(::ECCOMetadata{<:Any, <:ECCO2Daily})   = "https://ecco.jpl.nasa.gov/drive/files/ECCO2/cube92_latlon_quart_90S90N/daily/"
urls(::ECCOMetadata{<:Any, <:ECCO4Monthly}) = "https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/interp_monthly/"
```

**Comment by simone-silvestri** [comment]
ok, there seems to be a problem with Makie

**Comment by simone-silvestri** [comment]
Can we merge this? I think the problem with Makie is handled in #156

**Comment by glwagner** [comment]
I am so down to merge this

**Pull request merged**